cmake/GTClangSetupCUDA.cmake

```diff
@@ -0,0 +1,60 @@
+macro( gtclang_add_nvcc_flags )
+  foreach( flag ${ARGV} )
+    set( CUDA_NVCC_FLAGS ${CUDA_NVCC_FLAGS} ${flag} )
+  endforeach()
+endmacro()
+
+macro( gtclang_setup_CUDA )
+
+  find_package( CUDA 8.0 REQUIRED )
+  message( STATUS "CUDA detected: " ${CUDA_VERSION} )
+
+  set( CUDA_PROPAGATE_HOST_FLAGS ON )
+  set( CUDA_LIBRARIES "" ) # Is this required?
+
+  set( CUDA_ARCH "CUDA_ARCH-NOTFOUND" CACHE STRING "CUDA architecture (e.g. sm_35, sm_37, sm_60); precedence over GPU_DEVICE" )
+
+  if( NOT CUDA_ARCH )
```

Comment:
do we ever fail this condition? If we set it to CUDA_ARCH-NOTFOUND

Reply:
I was testing this, and actually this `if(NOT CUDA_ARCH)` evaluates to true if the variable is set to something like "XXXX-NOTFOUND" .... 
surprise, one of the magics of cmake...